lib/phoenix_live_view/helpers.ex

```diff
@@ -256,7 +264,7 @@ defmodule Phoenix.LiveView.Helpers do
 
   def __live_component__(%Socket{}, %{kind: :component, module: component}, assigns)
       when is_list(assigns) or is_map(assigns) do
-    assigns = assigns |> Map.new() |> Map.put_new(:id, nil)
+    assigns = assigns |> Map.new() |> Map.put_new(:id, nil) |> Map.put_new(:flash, %{})
```

Comment:
So I think we may have an issue here. If we allow the user to pass a flash, then the flash becomes a permanent state of that component, which means that every time we send an event to the component, the flash will show up the parent.

Similarly, if you do a `live_patch` from the component, the component may still exist after the patch, so we need to clear up the flash of the component so we don't keep adding the flash to the component forever and ever.

Therefore, I think instead of setting the flash here, we should set it elsewhere. If the only goal for flash in components is to pass state from child to parent, then we probably want to set it to an empty map on `configure_socket_for_component`. Another more conservative option is to set it to an empty map here, using `Map.put` instead, and then reset it back to an empty flash every time we call `handle_event ` on the component. I think I prefer the first approach.

Thoughts?